arkane/encorr/isodesmic.py

```diff
@@ -441,9 +817,18 @@ def multiple_error_canceling_reaction_search(self, n_reactions_max=20, milp_soft
                 for index in subset_indices:
                     subset_queue.append(np.delete(subset, index))
 
+                # Clean up the queue to remove subsets that would allow for already found reactions
+                new_queue = []
+                reaction_indices = [subset[i] for i in subset_indices]
+                for s in subset_queue:
+                    if not all([i in s for i in reaction_indices]):
+                        new_queue.append(s)
+
+                subset_queue = new_queue
+
```

Comment:
Thanks for the comment. My understanding is that it is to try to only use subsets of reference species that would not allow for already found reactions. I have re-written this function to improve readability.

Original:

```
        subset_queue = [np.arange(0, len(self.reference_species))]
        reaction_list = []

        while (len(subset_queue) != 0) and (len(reaction_list) < n_reactions_max):
            subset = subset_queue.pop()
            if len(subset) == 0:
                continue
            reaction, subset_indices = self._find_error_canceling_reaction(
                subset
            )
            if reaction is None:
                continue
            else:
                reaction_list.append(reaction)

                for index in subset_indices:
                    subset_queue.append(np.delete(subset, index))

                # Clean up the queue to remove subsets that would allow for already found reactions
                new_queue = []
                reaction_indices = [subset[i] for i in subset_indices]
                for s in subset_queue:
                    if not all([i in s for i in reaction_indices]):
                        new_queue.append(s)

                subset_queue = new_queue
```

My re-write:

```
        subset_queue = [np.arange(0, len(self.reference_species))] # indices of subsets of reference species
        reaction_list = []

        while subset_queue and len(reaction_list) < n_reactions_max:
            current_subset = subset_queue.pop()
            if len(current_subset) == 0:
                continue
            
            reaction, used_subset_indices = self._find_error_canceling_reaction(current_subset) # used_subset_indices are the indices of the current subset that were used in the reaction
            
            if reaction is None:
                continue
            
            reaction_list.append(reaction)

            # Generate new subsets by removing each of the used_subset_indices from the current subset
            for index in used_subset_indices:
                new_subset = np.delete(current_subset, index)
                subset_queue.append(new_subset)

            # Remove subsets that include all of the already used reference species
            used_ref_spc_indices = {current_subset[i] for i in used_subset_indices}
            subset_queue = [
                s for s in subset_queue
                if not used_ref_spc_indices.issubset(s)
            ]
```